### ISSUE TITLE
new lines and comments in rules.json

### ISSUE THREAD

**Issue by sebastianovide** [goldibex/targaryen#30]
Hi all,

we are evaluating how to test firebase rules and this seems to be the only real solution. Our rules.json is getting big, and to keep things clear we add comments and new lines using the same syntax accepted by firebase. Nevertheless targaryen complains about the json format. It doesn't like new lines nor comments in the json.

(Of course I could clean up the file before using targaryen.)

Is there any plans to support new lines and comments ?

thanks

**Comment by goldibex** [comment]
I wasn't actually aware that Firebase permits deviations from conventional
JSON in the format of the security rules. I'll look into this and get back
to you.

**Comment by SingaporeClouds** [comment]
Here are a few examples that I have seen in valid security rules. 
```
{
    "rules": {
        //anyone can read
        ".read": true,
        ".write": true, // anyone can write
        "users":{
         /* Even multiline comments work
          * TODO: Users should only be able to write their own records. 
          */
        }
    }
}
```

**Comment by alexweber** [comment]
@sebastianovide You can get a better dev experience using the [Firebase Bolt Compiler](https://github.com/firebase/bolt), there's a SublimeText bundle you can use in IDEA too for syntax highlighting.  I've found it makes it a bit easier to work with complex rules.

**Comment by dinoboff** [comment]
AFAIK it only affected the CLI (and support for comments is implemented in v2.2.0).

When using the API (i.e `targaryen.setFirebaseRules(someRule)`), targaryen doesn't load the rules and you can parse the rules with something supporting comments and new lines in strings. You can use something like json5 or even yaml if you don't mind converting it to json before uploading the rules.

**Comment by dinoboff** [comment]
AFAIK there's no node parser that support both comments and new lines in json string and writing a JSON parser is out of the scope of this project at this point.

If there's a parser available that support the same features than Firebase, fell free to reopen this issue.
